enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,608 @@
+previously configured and the cluster is upgraded, the default value `[]` for
+`Subnets` will clear the annotation for existing IngressControllers, which
+would break cluster ingress. However, requiring the service to be deleted to effectuate
+a `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+value different from the current value of the annotation prevents this
+automatic removal of the service annotation on upgrade.
+
+**Note**: Cluster admins upgrading with an unmanaged subnet annotation don't
+need to delete the service to propagate the subnet values, instead they can follow the
+[Unmanaged Subnet Annotation Migration Workflow](#unmanaged-subnet-annotation-migration-workflow).
+
+##### Reason #3: The CCM Doesn't Reconcile NLB Subnets Updates after creation.
+
+The CCM doesn't reconcile `service.beta.kubernetes.io/aws-load-balancer-subnets`
+annotation updates to NLBs once it has been created. However, requiring a service
+deletion will effectuate these changes.
+
+See [CCM Doesn't Reconcile NLB Subnets Updates](#ccm-doesnt-reconcile-nlb-subnets-updates)
+for more details on why the CCM doesn't reconcile these updates.
+
+### Install Config Proposal (Day 1)
+
+TODO: Install time configuration proposal
+
+### Implementation Details/Notes/Constraints
+
+When an IngressController is created with
+`spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+specified, the Ingress Operator will set the
+`service.beta.kubernetes.io/aws-load-balancer-subnets` annotation
+on the LoadBalancer-type Service.
+
+However, if `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+is updated on an existing IngressController, the Ingress Operator will NOT
+update `service.beta.kubernetes.io/aws-load-balancer-subnets` until the
+cluster admin deletes the LoadBalancer-type service as indicated by the
+`LoadBalancerProgressing` status (see [Effectuating Subnet Updates](#effectuating-subnet-updates)).
+When the Ingress Operator recreates the LoadBalancer-type service, it will
+then configure `service.beta.kubernetes.io/aws-load-balancer-subnets` with
+the new `Subnets` value.
+
+The `status.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+will eventually reflect the _actual_ subnet value by mirroring the value of
+`service.beta.kubernetes.io/aws-load-balancer-subnets` on the service. There are
+two scenarios in which the `Subnets` status won't be equal to the `Subnets` spec:
+
+1. The cluster admin manually configured the `service.beta.kubernetes.io/aws-load-balancer-subnets`
+   annotation on the service.
+2. The cluster admin configured the IngressController's `Subnets` spec, but hasn't
+   effectuated the change by deleting the service (see [Effectuating Subnet Updates](#effectuating-subnet-updates)).
+
+Note that this status usage is consistent with the approach in [LoadBalancer Allowed Source Ranges](/enhancements/ingress/lb-allowed-source-ranges.md),
```

Comment:
"status usage" is not defined.  Maybe just say:
```suggestion
Note that updating the status with admin instructions is consistent with the approach in [LoadBalancer Allowed Source Ranges](/enhancements/ingress/lb-allowed-source-ranges.md),
```

Reply:
Sorry, I took this update, but was incorrect. This isn't what this paragraph was explaining.

This isn't about the pattern of "admin instruction" `status.conditions`, this is about the `status.endpointPublishingStrategy`, which is an un-standardized field structure where past developers have been populating using 2-3 different patterns. Some `status.endpointPublishingStrategy` fields mean it's the "default", some mean it's what was set when the IngressController was first created, some means it's the 'effective' value.

We are trying to perpetuate the last one: `status.endpointPublishingStrategy` == effective value. No consistency is confusing for cluster admins, as they can't consistency interpret `status.endpontPublishingStrategy`. It also is confusing for developers debugging `status.endpontPublishingStrategy`. Miciah and I discussed this and he asked me add this paragraph to this EP.

I can see the confusion now. Let me refactor this to make it more clear that I am talking about `status.endpointPublishingStrategy`.